### DISCUSSION TITLE
fid remove

### DISCUSSION THREAD

**Discussion by petrparolek** [jiripudil/Naja#358] [General]
Hi, I have otfen troubles with Javascript..I haven't found anywhere how I can remove fid from URL after redirect.

I know about this for nette.ajax.js https://github.com/vojtech-dobes/nette.ajax.js/blob/master/extensions/fidRemove.ajax.js.

Could you anybody provide help please?

**Comment by jiripudil** [comment]
Hi, if it's a hard redirect, I think it's out of Naja's scope, you can alter the URL in your code, as suggested [here](https://forum.nette.org/cs/35109-naja-fid-remove-extension). If you're using Naja, I suggest calling `replaceState()` with `window.history.state` instead of an empty object, otherwise you could break some of Naja's features.

If it is a redirect handled by Naja, you can achieve this server-side by providing a [`postGet`](https://naja.js.org/#/history?id=prg-considerations) URL without the `_fid` parameter – Naja will then do the rest. Alternatively, I think you could also call that `replaceState()` above in a `success` listener.
